packages/eslint-plugin/src/rules/no-useless-template-literals.ts

```diff
@@ -53,6 +53,24 @@ export default createRule<[], MessageId>({
       return isString(type);
     }
 
+    function isLiteral(expression: TSESTree.Expression): boolean {
+      return expression.type === AST_NODE_TYPES.Literal;
+    }
+
+    function isInfinityIdentifier(expression: TSESTree.Expression): boolean {
+      return (
+        expression.type === AST_NODE_TYPES.Identifier &&
+        expression.name === 'Infinity'
+      );
+    }
+
+    function isNaNIdentifier(expression: TSESTree.Expression): boolean {
+      return (
+        expression.type === AST_NODE_TYPES.Identifier &&
+        expression.name === 'NaN'
+      );
+    }
```

Comment:
[Style] Nit: both of these functions check `expression.type === AST_NODE_TYPES.Identifier`. I think it'd be just a bit cleaner code to avoid the duplicate check and instead have a single function for both. WDYT?

This is a super nitpicky nit and definitely not a blocker 😄. 

A good potential followup might be to look into writing a general catch-all function for this. Searching `expression.type === AST_NODE_TYPES.Identifier` shows a few dozen results. Maybe half or more of them or so are similar calls.

Reply:
Yeah, so I basically took inspiration from this:
https://github.com/typescript-eslint/typescript-eslint/blob/8622286207b3e04ddd260eaa8ac71927d47edd7d/packages/eslint-plugin/src/rules/no-inferrable-types.ts#L73-L80

And thought about extracting it to a util, WDYT?

Reply:
Probably useful to have that as a util yeah. I'm not super picky here to be honest.